service/history/replication/task_ack_manager_test.go

```diff
@@ -298,7 +311,7 @@ func TestTaskAckManager_GetTasks(t *testing.T) {
 				LastRetrievedMessageID: 11,
 				HasMore:                false,
 			},
-			config: &config.Config{MaxResponseSize: testMaxResponseSize},
+			config: testConfig,
```

Comment:
consider introducing a `newDynamicTaskBatchSizerFn()` and override it in tests to create new test cases.